src/Accelerators/Accelerator.hpp

```diff
@@ -24,8 +24,17 @@ namespace accel {
 
 class Accelerator {
 public:
-  Accelerator();
+  /// Kinds of accelerators.
+  enum class Kind {
+    NNPA, // IBM Telum coprocessor
```

Comment:
This exposes the accelerator names outside of their subdirectory. My idea for this is to have CMake generate the enum values in an include and so on ...

Reply:
This is just an enumerator so I do not understand the reason we do not want to "expose it".  It does not force the accelerator class type to be exposed.

Reply:
Yes, but in my opinion, it is simpler to contain everything in the specific accelerator subdirectory and have CMake generate the "glue" based on the values specified in the environment variable. I can do it if you want, but if you feel strongly that this way is better I guess I can let it slide.

Reply:
@caoimhinuibrian do you have the cmake code to generate the include file you mentioned, happy to add it to the patch.